FilePersistence/src/version_control/Diff3Parse.cpp

```diff
@@ -162,6 +161,48 @@ QList<Model::NodeIdType> Diff3Parse::longestCommonSubsequence(const QList<Model:
 	return lcs;
 }
 
+QList<Model::NodeIdType> Diff3Parse::longestCommonSubsequence3(const QList<Model::NodeIdType> listA,
+																				const QList<Model::NodeIdType> listB,
+																				  const  QList<Model::NodeIdType> listC)
+{
+	int m = listA.size() + 1;
+	int n = listB.size() + 1;
+	int o = listC.size() + 1;
+
+	int*** lcsLength = new int**[m];
+	for (int i = 0; i < m; ++i)
+	{
+		lcsLength[i] = new int*[n];
+		for (int j = 0; j < n; ++j)
+			lcsLength[i][j] = new int[o];
+	}
+
+	for (int i = 0; i < m; ++i)
+		for (int j = 0; j < n; ++j)
+			for (int k = 0; k < o; ++k)
+				lcsLength[i][j][k] = 0;
+
+	for (int i = 1; i < m; ++i)
+		for (int j = 1; j < n; ++j)
+			for (int k = 1; k < o; ++k)
+				if ((listA.at(i-1) == listB.at(j-1)) && (listA.at(i-1) == listC.at(k-1)))
+					lcsLength[i][j][k] = lcsLength[i-1][j-1][k-1] + 1;
+				else
+					lcsLength[i][j][k] = std::max(lcsLength[i-1][j][k], std::max(lcsLength[i][j-1][k], lcsLength[i][j][k-1]));
+
+	QList<Model::NodeIdType> lcs = backtrackLCS3(lcsLength, listA, listB, listC, listA.size(), listB.size(), listC.size());
+
+	for (int i = 0; i < m; ++i)
+		for (int j = 0; j < n; ++j)
+			delete[] lcsLength[i][j];
+	for (int i = 0; i < m; ++i)
+		delete[] lcsLength[i];
```

Comment:
instead of having an extra loop here, you can insert this delete after the inner loop above (outside the inner loop)